.github/workflows/doxygen.yml

```diff
@@ -7,8 +7,9 @@ on:
 
 jobs:
   upload-doc:
+    if: ${{ github.repository == 'p4lang/behavioral-model' && github.ref == 'refs/heads/main' }}
```

Comment:
I think the `github.ref` is redundant with the workflow definition, as the workflow only runs for the main branch already. But it is harmless.